lib/modules/ai_bot/commands/search_command.rb

```diff
@@ -154,17 +154,33 @@ def process(**search_args)
       end
 
       @last_num_results = posts.length
+      # this is the general pattern from core
+      # if there are millions of hidden tags it may fail
+      hidden_tags = nil
 
       if posts.blank?
         { args: search_args, rows: [], instruction: "nothing was found, expand your search" }
       else
         format_results(posts, args: search_args) do |post|
-          {
+          category_names = [
+            post.topic.category&.parent_category&.name,
+            post.topic.category&.name,
+          ].compact.join(" > ")
+          row = {
             title: post.topic.title,
             url: Discourse.base_path + post.url,
             excerpt: post.excerpt,
             created: post.created_at,
+            category: category_names,
           }
+
+          if SiteSetting.tagging_enabled
+            hidden_tags ||= DiscourseTagging.hidden_tag_names
+            # using map over pluck to avoid n+1 (assuming caller preloading)
```

Comment:
Instead of assuming can we check whether the association is loaded or not? like `post.topic.association(:tags).loaded?`

Reply:
I think it is the responsibility of Search or SemanticSearch ... but maybe we can raise a warning in case this happens.